compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -328,7 +328,31 @@ impl<'tcx> Collector<'tcx> {
                         .map(|child_item| self.build_dll_import(abi, child_item))
                         .collect()
                 }
-                _ => Vec::new(),
+                _ => {
+                    for child_item in foreign_mod_items {
+                        if self.tcx.def_kind(child_item.id.def_id).has_codegen_attrs()
```

Comment:
I think `has_codegen_attrs` should always be true for any child_item here by construction. We already just call `codegen_fn_attrs` without gating it on that check in the `build_dll_import` call above.

Reply:
Not quite: we're walking through everything withing a `ForeignMod`, that includes `ForeignTy` which doesn't have codegen attrs (this is why I added the `has_codegen_attrs` check).